app/src/main/java/org/odk/odknotifications/Activities/MainActivity.java

```diff
@@ -293,9 +249,40 @@ public void onBackPressed() {
         }
     }
 
+    private void addNotifications(){
```

Comment:
We shouldn't be doing the initialisation of all these things again and again. Its better we initialize all these things in `onCreate` and then update the list which was passed in the adapter and then call `notifyDataSetChanged`

Reply:
Done.